crates/proto/Cargo.toml

```diff
@@ -42,8 +42,8 @@ dns-over-rustls = ["dns-over-tls", "rustls", "rustls-pemfile", "tokio-rustls", "
 dns-over-native-tls = ["dns-over-tls", "native-tls", "tokio-native-tls"]
 dns-over-openssl = ["dns-over-tls", "openssl", "tokio-openssl"]
 
-dns-over-https-rustls = ["dns-over-https", "dns-over-rustls", "webpki-roots"]
-dns-over-https = ["bytes", "dns-over-tls", "h2", "http"]
+dns-over-https-rustls = ["dns-over-https"]
+dns-over-https = ["bytes", "h2", "http", "dns-over-rustls", "webpki-roots", "tokio-runtime"]
```

Comment:
This doesn't look right to me, I think `dns-over-https` should only have TLS-implementation-neutral dependencies.

Reply:
Well, in practice, the https implementation in trust-dns-resolver depends on rustls and is tightly coupled with it.
If `dns-over-https` is the only feature enabled, then the compiler would give error on missing imports from `rustls`.

Reply:
Hmm, okay, in that case this probably makes sense.